skglm/datafits/single_task.py

```diff
@@ -774,3 +773,114 @@ def _get_indptr(self, vals, indices):
         indptr.append(n_indices)
 
         return np.asarray(indptr, dtype=np.int64)
+
+
+class WeightedQuadratic(BaseDatafit):
+    """Weighted Quadratic datafit.
+
+    The datafit reads:
+
+    .. math:: 1 / (2 xx  n_"samples") ||y - Xw||_2 ^ 2
+
+    Attributes
+    ----------
+    Xty : array, shape (n_features,)
+        Pre-computed quantity used during the gradient evaluation.
+        Equal to ``X.T @ y``.
+    sample_weights : array, shape (n_train_samples,)
+        Weights for each sample.
+    Note
+    ----
+    The class is jit compiled at fit time using Numba compiler.
+    This allows for faster computations.
+    """
+
+    def __init__(self, sample_weights):
+        self.sample_weights = sample_weights
+        pass
+
+    def get_spec(self):
+        spec = (
+            ('Xty', float64[:]),
+            ('sample_weights', float64[:]),
+        )
+        return spec
+
+    def params_to_dict(self):
+        return {'sample_weights':self.sample_weights}
```

Comment:
missing whitespace after `:`. You can add a linter in vscode to do this automatically for you whenever you save